### PR TITLE
Use correct effective date for GW holiday stops

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This change ensures that the negative charge is applied to the next invoice by using the charge-through date (the first day of the next service period) as the effective date.

### CODE REVIEW

File: handlers/holiday-stop-processor/src/main/scala/com/gu/holidaystopprocessor/SubscriptionUpdate.scala [hunk 1]
```diff
@@ -15,42 +15,32 @@ case class SubscriptionUpdate(currentTerm: Option[Int], add: Seq[Add]) {
...
-          customerAcceptanceDate = effectiveDate,
-          serviceActivationDate = effectiveDate,
-          chargeOverrides = Seq(
-            ChargeOverride(
-              productRatePlanChargeId = config.holidayCreditProductRatePlanChargeId,
-              HolidayStart__c = stoppedPublicationDate,
-              HolidayEnd__c = stoppedPublicationDate,
-              price = HolidayCredit(subscription)
+  ): Either[HolidayStopFailure, SubscriptionUpdate] = {
+    subscription.originalRatePlanCharge.flatMap(_.chargedThroughDate)
```
Comment (paulbrown1982) [inline handlers/holiday-stop-processor/src/main/scala/com/gu/holidaystopprocessor/SubscriptionUpdate.scala]: Does chargedThroughDate also need `.plusDays(1)`?

Comment (kelvin-chappell) [reply handlers/holiday-stop-processor/src/main/scala/com/gu/holidaystopprocessor/SubscriptionUpdate.scala]: No, the chargedThroughDate is the first date of the next billing period.  I've checked the invoice previews with @mario-galic and it does the right thing, ie. holiday stops taking place in this billing period appear in the next invoice.  There's a separate card for considering what to do if holiday stop goes over threshold into the next billing period.

Comment (mario-galic) [reply handlers/holiday-stop-processor/src/main/scala/com/gu/holidaystopprocessor/SubscriptionUpdate.scala]: Note the difference between `Service period` shown on an invoice, and corresponding API dates `[effectiveStartDate, effectiveEndDate)`:
 
* Service Period: 20.10.2018-19.01.2019
* `[effectiveStartDate, effectiveEndDate)`: 20.10.2018-20.01.2019

On an invoice the end date is _inclusive_ whist on an API response it is _exclusive_

Comment (kelvin-chappell) [reply handlers/holiday-stop-processor/src/main/scala/com/gu/holidaystopprocessor/SubscriptionUpdate.scala]: I think it's consistent.  The `chargedThroughDate` is the first day in the next service period if you're paying in advance.

**Pull request merged**